daemon/gamemode-dbus.c

```diff
@@ -189,6 +191,84 @@ static int method_query_status_by_pid(sd_bus_message *m, void *userdata,
 	return sd_bus_reply_method_return(m, "i", status);
 }
 
+/**
+ * Handles the RegisterGameByPID D-BUS Method
+ */
+static int method_register_game_by_pidfd(sd_bus_message *m, void *userdata,
+                                         __attribute__((unused)) sd_bus_error *ret_error)
+{
+	int fds[2] = { -1, -1 };
+	pid_t pids[2] = { 0, 0 };
+	GameModeContext *context = userdata;
+
+	int ret = sd_bus_message_read(m, "hh", &fds[0], &fds[1]);
+	if (ret < 0) {
+		LOG_ERROR("Failed to parse input parameters: %s\n", strerror(-ret));
+		return ret;
+	}
+
+	int reply = pidfds_to_pids(fds, pids, 2);
```

Comment:
Are the FDs being leaked here? I don't see a close for them. Same for the other methods.

Reply:
I believe `sd_bus_message_read` won't `dup` the file descriptors so they are being closed when the message is freed. Let me verify this though. The mapage is silent on it :/

Reply:
Confirmed with @poettering, doesn't dup, we should be good.